subsys/net/ip/net_shell.c

```diff
@@ -2659,24 +2659,52 @@ static enum net_verdict handle_ipv6_echo_reply(struct net_pkt *pkt,
 						struct net_ipv6_hdr *ip_hdr,
 						struct net_icmp_hdr *icmp_hdr)
 {
-	ARG_UNUSED(icmp_hdr);
+	NET_PKT_DATA_ACCESS_CONTIGUOUS_DEFINE(icmp_access,
+					      struct net_icmpv6_echo_req);
+	u32_t cycles;
+	struct net_icmpv6_echo_req *icmp_echo;
 
-	PR_SHELL(shell_for_ping, "Received echo reply from %s to %s\n",
+	icmp_echo = (struct net_icmpv6_echo_req *)net_pkt_get_data(pkt, &icmp_access);
+	if (icmp_echo == NULL) {
+		return -NET_DROP;
+	}
+
+	net_pkt_skip(pkt, sizeof(*icmp_echo));
+	if (net_pkt_read_be32(pkt, &cycles)) {
+		return -NET_DROP;
+	}
+
+	cycles = k_cycle_get_32() - cycles;
+
+	PR_SHELL(shell_for_ping, "%d bytes from %s to %s: icmp_seq=%d ttl=%d "
+#ifdef CONFIG_IEEE802154
+		 "rssi=%d "
+#endif
+		 "time=%.2f ms\n",
+		 ntohs(ip_hdr->len) - net_pkt_ipv6_ext_len(pkt) - NET_ICMPH_LEN,
 		 net_sprint_ipv6_addr(&ip_hdr->src),
-		 net_sprint_ipv6_addr(&ip_hdr->dst));
+		 net_sprint_ipv6_addr(&ip_hdr->dst),
+		 ntohs(icmp_echo->sequence),
+		 ip_hdr->hop_limit,
+#ifdef CONFIG_IEEE802154
+		 net_pkt_ieee802154_rssi(pkt),
+#endif
+		 SYS_CLOCK_HW_CYCLES_TO_NS(cycles) / 1000000.f);
 	k_sem_give(&ping_timeout);
-	remove_ipv6_ping_handler();
```

Comment:
remove_ipv4_ping_handler should be removed also.

Reply:
good catch!